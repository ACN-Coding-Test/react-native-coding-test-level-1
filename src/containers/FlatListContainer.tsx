import React from 'react';
import {Dimensions, ScrollView, StatusBar, View} from 'react-native';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
import {Header} from '../components';

const width = Dimensions.get('screen').width;
const height = Dimensions.get('screen').width;

type Props = {
    backgroundColor: string;
    children: React.ReactNode;
    title: string;
    statusBarColor: string;
    isMain?: boolean;
};

const Container = ({
    children,
    backgroundColor,
    title,
    statusBarColor,
    isMain
}: Props) => {
    return (
        <View
            style={{
                flex: 1,
                width,
                height,
                backgroundColor: backgroundColor
            }}>
            <StatusBar
                backgroundColor={backgroundColor}
                barStyle="light-content"
            />
            <Header
                title={title}
                statusBarColor="red"
                backgroundColor="white"
                isMain={isMain}
            />
            <View
                style={{
                    flex: 1
                }}>
                {children}
            </View>
        </View>
    );
};

export default Container;
