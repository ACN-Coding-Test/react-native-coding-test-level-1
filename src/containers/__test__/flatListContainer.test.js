import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';

import FlatListContainer from '../FlatListContainer';

test('renders Loading correctly', () => {
    const tree = renderer
        .create(
            <FlatListContainer
                backgroundColor="white"
                statusBarColor="red"
                title="PokeDex"
                isMain={true}>
                <View style={{flex: 1}}></View>
            </FlatListContainer>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
