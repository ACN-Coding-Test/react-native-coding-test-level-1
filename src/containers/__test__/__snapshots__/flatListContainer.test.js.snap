// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "height": 750,
      "width": 750,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "height": 70,
      }
    }
    testID="header"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": "5%",
          "paddingRight": "5%",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="back-button"
        >
          <Text
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontSize": 16,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "marginVertical": 2,
                    },
                    {
                      "letterSpacing": 1.5,
                    },
                  ],
                ],
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flex": 3,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "letterSpacing": 1.5,
              "textAlign": "center",
            }
          }
        >
          PokeDex
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
